components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/AddCmd.java

```diff
@@ -430,11 +435,62 @@ else if(typeAndProjectName[0].equals("route")){
                     GatewayCmdUtils.saveConfig(newConfig, toolkitConfigPath);
                 }
             }
+        }//todo: implement add route command
+        else {
+            if (resource_id.isEmpty()) {
```

Comment:
Here you can use &&.

Reply:
we can't add like that, it changes the logic.

Reply:
because "else" is implemented on whether this command is add api or add route, and the second "if" is just to identify whether resource_id is provided or not. Serves two different purposes.

Reply:
ok